File: src/main/java/com/zhongan/devpilot/completions/inline/render/DevPilotInlay.java
//
// Source code recreated from a .class file by IntelliJ IDEA
// (powered by FernFlower decompiler)
//

package com.zhongan.devpilot.completions.inline.render;

import com.intellij.openapi.Disposable;
import com.intellij.openapi.editor.Editor;
import com.zhongan.devpilot.completions.prediction.DevPilotCompletion;
import java.awt.Rectangle;

public interface DevPilotInlay extends Disposable {
    static DevPilotInlay create(Disposable parent) {
        return new DefaultDevPilotInlay(parent);
    }

    Integer getOffset();

    Boolean isEmpty();

    Rectangle getBounds();

    void render(Editor var1, DevPilotCompletion var2, int var3);
}

File: src/main/java/com/zhongan/devpilot/completions/common/inline/render/DevPilotInlay.kt
package com.zhongan.devpilot.completions.common.inline.render

import com.intellij.openapi.Disposable
import com.intellij.openapi.editor.Editor
import com.zhongan.devpilot.completions.common.prediction.DevPilotCompletion
import java.awt.Rectangle

interface DevPilotInlay : Disposable {
    val offset: Int?
    val isEmpty: Boolean

    fun getBounds(): Rectangle?
    fun render(editor: Editor, completion: DevPilotCompletion, offset: Int)

    companion object {
        @JvmStatic
        fun create(parent: Disposable): DevPilotInlay {
            return DefaultDevPilotInlay(parent)
        }
    }
}
